### ISSUE TITLE
Site doesn't load

### ISSUE THREAD

**Issue by MutenR0sh1** [fhuhne/CR-Unblocker#20]
After the Crunchyroll maintenance a few hours ago I can't open the website with enabled switch region. The site will load infinitely. 
When switch region is disabled, the website will open normally.

**Comment by fhuhne** [comment]
Unrelated to maintenance, proxy lost connectivity. Should be back up now :)

**Comment by Jules-A** [comment]
Right now I just keep getting "There was an issue loading the page. Please refresh it." followed by a redirect to log in which just keeps looping back with the same error. Is that also just the proxy being now?

**Comment by fhuhne** [comment]
Yes, it seems like it has some problems again. Seems like I have to investigate this issue some more as it becomes more common over time.

**Comment by Jules-A** [comment]
> Yes, it seems like it has some problems again. Seems like I have to investigate this issue some more as it becomes more common over time.

@fhuhne Is it possible for the extension to detect proxy failures and fallback to not using it (or an option to at least) instead of an unusable site? Seems to keep happening every now and then and when it does it's rather annoying having to disable the extension.

**Comment by fhuhne** [comment]
Would love to have a solution like this! I checked this a while ago to find a solution but found nothing usable.

There is an proxy.onError API (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/onError) but it seems like that only handles incorrect proxy configuration and no error handling from the response.

Looking at the available options for proxy.onRequest again, I found failoverTimeout. Maybe this will help? Will do a small release this weekend including this option and a "direct" proxy configuration as fallback. Hopefully that will at least give us a usable site. Also updated a script on the proxy server which might have been the culprit for the last couple of downtimes.

**Comment by JicuNull** [comment]
The proxy seems to be offline again. Since a few hours I can't load the page with activated addon.

**Comment by fhuhne** [comment]
Hi all, some improvements added as of today:

* Release 3.0.2 adds a fallback. When the proxy server is not responding properly it will be bypassed. This will effectively disable the addon during downtimes.
* I improved my monitoring. Until now I only checked if the ports are up, but not if the server has outgoing connectivity. My monitoring now includes a rule for that.
* I noticed the connection dropped mostly after an uptime of a bit over 1 month. I assume it is some kind of timeout. I set the service to restart periodically to prevent this in the future.

I hope this solves this kind of downtime in the future :)
